model/blueprint.py

```diff
@@ -1,59 +1,30 @@
-    image_id = columns.Text(required=False, max_length=500)
-    vpc_id = columns.Text(required=False, max_length=100)
-    subnet_id = columns.Text(required=False, max_length=200)
-    public_route = columns.Boolean(required=False)
-    ig_id = columns.Text(required=False, max_length=100)
-    route_table = columns.Text(required=False, max_length=100)
-    vm_id = columns.Text(required=False, max_length=200)
-    project = columns.Text(primary_key=True, max_length=50)
-    nic_id = columns.Text(max_length=200)
-    disk_clone = columns.List(value_type=columns.Map(key_type=columns.Text(),value_type=columns.Text()))
-
-
-
-class BluePrintMongo(Document):
-    host = StringField(required=True, max_length=200)
-    ip = StringField(required=True)
-    ip_created = BooleanField(required=True, default=False)
-    subnet = StringField(required=True, max_length=50)
-    network = StringField(required=True, max_length=50)
-    ports = ListField()
-    cores = StringField(max_length=2)
-    cpu_model = StringField(required=True, max_length=150)
-    ram = StringField(required=True, max_length=50)
-    machine_type = StringField(required=True, max_length=150)
-    status = StringField(required=False, max_length=100)
-    image_id = StringField(required=False, max_length=500)
-    vpc_id = StringField(required=False, max_length=100)
-    subnet_id = StringField(required=False, max_length=200)
-    public_route = BooleanField(required=False)
-    ig_id = StringField(required=False, max_length=100)
-    route_table = StringField(required=False, max_length=100)
-    vm_id = StringField(required=False, max_length=200)
-    project = StringField(required=True, max_length=50)
-    nic_id = StringField(max_length=200)
-    disk_clone = ListField()
-    meta = {
-        'indexes': [
-            {'fields': ('host', 'project'), 'unique': True}
-        ]
-    }
+from utils.database import Base
+from sqlalchemy import Column, String, Boolean, ForeignKey
+from sqlalchemy.dialects.postgresql import ARRAY, JSON
 
 
+class Blueprint(Base):
+    
+    __tablename__ = 'blueprint'
+    
+    project = Column(String, ForeignKey("project.name", onupdate='CASCADE', ondelete='CASCADE'), primary_key=True, unique=True)
+    host = Column(String, primary_key=True)
+    ip = Column(String, nullable=False)
+    ip_created = Column(Boolean, nullable=False, default=False)
+    subnet = Column(String, nullable=False)
+    network = Column(String, nullable=False)
+    ports = Column(ARRAY(String))
+    cores = Column(String)
+    cpu_model = Column(String, nullable=False)
+    ram = Column(String, nullable=False)
+    machine_type = Column(String, nullable=False)
+    status = Column(String)
+    image_id = Column(String)
+    vpc_id = Column(String)
+    subnet_id = Column(String)
+    public_route = Column(Boolean)
+    ig_id = Column(String)
+    route_table = Column(String)
+    vm_id = Column(String)
+    nic_id = Column(String)
+    disk_clone = Column(ARRAY(JSON(String)))
```

Comment:
disk_clone status could be moved to disk_status table with a foreign key relation to the id of this blueprint table

Reply:
I think we could use disk table here.